src/api/File.js

```diff
@@ -78,29 +79,71 @@ class File extends Item {
         this.successHandler(`${downloadBaseUrl}?${downloadUrlQuery}`);
     }
 
+    /**
+     * Polls an item's infoUrl until it's representation is generated or an error occurrs.
+     * Calls callback if representation was successfully generated.
+     *
+     * @param {BoxItem} item - BoxItem that needs infoUrl polled
+     * @param {string} thumbanilUrl - thumbanil url of item
+     * @param {string, string) => void} callback - function to call if polling results in success
+     * @return {Promise<void>}
+     */
+    async pollInfoUrl(
```

Comment:
Might make sense to generalize this a bit more. Maybe something like `getRepresentation(item, type, polling)` that will get the representation type and do some sort of polling if the rep is not ready and polling is desired